Maintenance: support phpunit v12 & php-text-template v5

icarus94 opened the pull request:
- support new version `phpunit/php-text-template` (^5);
- support new version `phpunit/phpunit` (^12.0);
- add attribute for each annotation to support phpunit 12;
- use MockBuilder for building mocks instead of removed `getMockForAbstractClass` method

icarus94 committed:
- support new version `phpunit/php-text-template` (^5);
- support new version `phpunit/phpunit` (^12.0);
- add attribute for each annotation to support phpunit 12;
- use MockBuilder for building mocks instead of removed `getMockForAbstractClass` method

michalbundyra committed:
Merge branch 'master' into feat/support-phpunit-12

michalbundyra commented:
@icarus94 thanks for the PR, looks like some test on old versions are failing. Would you be able to have a look and try to sort it out? If not, I'll try to have a look during the week. Thanks 🙏

icarus94 committed:
- `getMockForAbstractClass` is removed from PhpUnit 12, using `MockBuilder` as suggested alternative
- use `MockBuilder::onlyMethods` on PhpUnit 8.3.0+ and `MockBuilder::setMethods` up to PhpUnit 8.3.0

icarus94 committed:
Merge remote-tracking branch 'origin/feat/support-phpunit-12' into feat/support-phpunit-12

michalbundyra commented:
Thanks, @icarus94 !
